gengokit/handler/templates.go

```diff
@@ -48,3 +48,23 @@ type {{.PackageName}}Service struct{}
 	{{end}}
 {{- end}}
 `
+const hookTempl = `
+package handler
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func InterruptHandler(errc chan<- error) {
```

Comment:
I feel like the name distracts from the idea that this will handle shutdown. `Shutdown` is a fine name.

Ideally, though, a "fill in the blank" function for the user should be basically empty. This `Shutdown` function should call an empty function that the the user can populate. The user should not have to maneuver between generated code that's crucial to proper operation, and the blank canvas of adding code. I'd go so far as to say they probably shouldn't even be in the same file. What I had envisioned was something like this.

**hooks.go**
```go
// Shutdown performs any user-defined cleanup tasks that need to be done
// before shutting down the server, like closing connections and any other 
// resource releasing.
func Shutdown() {
}
```
**main.go** (or somewhere else)
```
quitCh := make(chan os.Signal, 1)
signal.Notify(quitCh, syscall.SIGINT, syscall.SIGTERM)
go func() {
    sig := <-quitCh
    hooks.Shutdown()
    errc <- fmt.Errorf("%s", sig)
}()
```
Or even, more logically, I think, since I do not associate shutdown explicitly with signal handling:
```go
func main() {
    defer func() {
        hooks.Shutdown()
        log.Println("exiting")
    }()
    // ...
    go func() {
        errc <- fmt.Errorf("%s", <-quitCh)
    }()
    // ...
    // errc receive allows program execution to flow to the end of the file.
    log.Println(<-errc)
}
// EOF
```